source/common/http/utility.cc

```diff
@@ -100,6 +100,21 @@ std::string Utility::parseCookieValue(const HeaderMap& headers, const std::strin
   return state.ret_;
 }
 
+void Utility::removeCookieValues(const HeaderMap& headers, const std::string& key) {
+  headers.iterate([](const HeaderEntry& header) -> void {
+    if (header.key() == Http::Headers::get().Cookie.get().c_str()) {
+      for (const std::string& s : StringUtil::split(std::string{header.value().c_str()}, ';')) {
+        size_t first_non_space = s.find_first_not_of(" ");
+        size_t equals_index = s.find('=');
+        std::string k = s.substr(first_non_space, equals_index - first_non_space);
+        if (k == state->key_) {
```

Comment:
Yes, cookie headers can have many key value pairs. When I wrote this I thought the typical formatting was one k:v per header (similar to set cookie where there's metadata like Secure and HttpOnly), but that is incorrect.

I'll try to strip the specific cookie out instead.